### ISSUE TITLE
Linking and remembering open file path to save or export file path

### ISSUE THREAD

**Issue by Patowhiz** [africanmathsinitiative/R-Instat#6035]
Currently, if I open a file using **File > Open From File**, when saving the same file the Save Dialog prompt takes me to the documents folder. 
When I restart R-Instat, and use **File > Open From File** it always takes me to Document folder.

@rdstern and I would like to propose the following.

1. Can we make by default the Save and Export dialogs open the directory used in **Open From File** by user. 
2. Can we make R-Instat remember last opened file path similar to the way it remembers the recent files 

The above 2 functionalities would be similar to what a spreadsheet application like Excel does.

**Comment by rdstern** [comment]
This started as a discussion about the idea of a "working directory".  But I am happy to emulate what Excel does.  I note that R-Instat does open files from a variety of sources.  
a) The example by @Patowhiz is the most common.  We use File > Open from File and choose a file.  It remembers the directory.  Then Save or Export will start, by default in the same directory.
b) We re-use a recent file.  This counts the same as File > Open from File
c) If we use File  > Open From Library.  Then this doesn't affect the default directory.  But the save and export would still remember their last setting.
d) What happens after using Open and tidy netcdf files? Not sure here.  I am inclined that is not treated the same as Open from File.  So it would remember the directory in case it is used again, but does not affect the save or export?
e) Are there other situations?

I am quite keen to see what this looks like so happy for the work to be done if it is small.

**Comment by Patowhiz** [comment]
@dannyparsons what are your thoughts on this. Any suggestions?

**Comment by dannyparsons** [comment]
Yes this would be good to sort out, and as Roger mentions its not simple because some locations that we open we do not want to be remembered. But for the others it may be good to be remembered. But some things are not clear to me, for example saving a graph or the log file, does that remember the same global location or it's own? We also have a principle that dialogs open as they were closed so this may affect our thinking.

An option for a working directory might be sensible. We first need to define what that would mean.

If we can start defining what we want then we can get towards a solution.

**Comment by Patowhiz** [comment]
@dannyparsons I think a "working directory" is an advanced solution to the simple problem that I have defined above. A "working directory" to me is basically a directory that a user chooses to let a software automatically save files that are currently opened. In other software, even the metadata files(like settings, files about current opened files etc) and temporary files( like periodically auto saved data ) created are saved in the "working directory". I see the concept of working directory used mainly in software where a user works with more than 1 file and the files relate to each other to form a "project" e.g IDEs.

Wikipedia defines it as;

> In computing, the working directory of a process is a directory of a hierarchical file system, if any, dynamically associated with each process. It is sometimes called the current working directory

The problem defined, simply put is;
I have opened a file and now I want to save it. Should the software take me to Documents directory or take me to the directory from where I opened the last file. 

As @rdstern mentioned above, we can exclude special directories like "Open From Library" in the saving files feature or simply detect when the user tries to save in it and give warning or deny the permission.

@dannyparsons an important concern that I had not thought of is the principle of dialogs remembering the last state. This feature conflicts with it.
So I think the question then is, 
1. Should we neglect this simple feature and think more on the "working directory" and what it means in R-Instat
2. Or exempt the save dialogs from that principle
3. Or adapt this feature to that principle. For instance going a step further and displaying "Recent Folders" , see Excel example below. We still get to retain the principle while making it easy for the user to quickly save his/her files

Excel sample screenshot
<img width="433" alt="excel_sample" src="https://user-images.githubusercontent.com/21179007/96885253-17f6ec00-148b-11eb-967a-0724028de633.PNG">
